sql/core/src/main/scala/org/apache/spark/sql/execution/AggregatingAccumulator.scala

```diff
@@ -95,13 +95,13 @@ class AggregatingAccumulator private(
 
   /**
    * Driver side operations like `merge` and `value` are executed in the DAGScheduler thread. This
-   * thread does not have a SQL configuration so we attach our own here. Note that we can't (and
-   * shouldn't) call `merge` or `value` on an accumulator originating from an executor so we just
-   * return a default value here.
+   * thread does not have a SQL configuration so we attach our own here.
    */
   private[this] def withSQLConf[T](default: => T)(body: => T): T = {
```

Comment:
I think we should still forbid calling `value` on the executor side. How about
```
def withSQLConf[T](canRunOnExecutor: Boolean, default: => T)(body: => T): T = {
  if (conf != null) {
    // When we reach here we are on the driver side
    SQLConf.withExistingConf(conf)(body)
  } else if (canRunOnExecutor) {
    body
  } else {
    default
  }
}
```

Reply:
I see. Then, we don't need to check the condition `TaskContext.get != null`.